### PR TITLE
chore(ReactDomComponent): refactor code

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Refactor code to a more concise version.

### CODE REVIEW

File: packages/react-dom/src/client/ReactDOMComponent.js [hunk 1]
```diff
@@ -340,11 +346,7 @@ function setInitialDOMProperties(
       } else if (typeof nextProp === 'number') {
         setTextContent(domElement, '' + nextProp);
       }
-    } else if (
-      (enableFlareAPI && propKey === LISTENERS) ||
-      propKey === SUPPRESS_CONTENT_EDITABLE_WARNING ||
-      propKey === SUPPRESS_HYDRATION_WARNING
-    ) {
+    } else if (enableFlareAPIIsSet(propKey)) {
```
Comment (trueadm) [inline packages/react-dom/src/client/ReactDOMComponent.js]: These aren't all Flare APIs, so the naming is off.

File: packages/react-dom/src/client/ReactDOMComponent.js [hunk 2]
```diff
@@ -1045,14 +1039,10 @@ export function diffHydratedProperties(
...
-        (enableFlareAPI && propKey === LISTENERS) ||
-        propKey === SUPPRESS_CONTENT_EDITABLE_WARNING ||
-        propKey === SUPPRESS_HYDRATION_WARNING ||
+        enableFlareAPIIsSet(propKey) ||
         // Controlled attributes are not validated
         // TODO: Only ignore them on controlled tags.
-        propKey === 'value' ||
-        propKey === 'checked' ||
-        propKey === 'selected'
+        ['value', 'checked', 'selected'].includes(propKey)
```
Comment (trueadm) [inline packages/react-dom/src/client/ReactDOMComponent.js]: This is far less efficient than using simple or conditions.

Comment (jamesgeorge007) [reply packages/react-dom/src/client/ReactDOMComponent.js]: should I revert?

File: packages/react-dom/src/client/ReactDOMComponent.js (hunk 1 above)
Comment (jamesgeorge007) [reply packages/react-dom/src/client/ReactDOMComponent.js]: would you be kind to suggest a name for the respective variable?

Comment (jamesgeorge007) [reply packages/react-dom/src/client/ReactDOMComponent.js]: @trueadm ?

**Comment by jamesgeorge007** [comment]
@trueadm could you give me a better picture regarding the CI failure :thinking:

**Comment by trueadm** [comment]
Sure thing. I’m just home now so I’ll have a look and add more feedback tomorrow when I’m in the office :)

File: packages/react-dom/src/client/ReactDOMComponent.js (hunk 2 above)
Comment (trueadm) [reply packages/react-dom/src/client/ReactDOMComponent.js]: Yep

File: packages/react-dom/src/client/ReactDOMComponent.js (hunk 1 above)
Comment (trueadm) [reply packages/react-dom/src/client/ReactDOMComponent.js]: `isInternalPropDefinition`?

File: packages/react-dom/src/client/ReactDOMComponent.js [hunk 3]
```diff
@@ -697,11 +701,7 @@ export function diffProperties(
       }
     } else if (propKey === DANGEROUSLY_SET_INNER_HTML || propKey === CHILDREN) {
       // Noop. This is handled by the clear text mechanism.
-    } else if (
-      (enableFlareAPI && propKey === LISTENERS) ||
-      propKey === SUPPRESS_CONTENT_EDITABLE_WARNING ||
-      propKey === SUPPRESS_HYDRATION_WARNING
-    ) {
+    } else if (enableFlareAPI(propKey)) {
```
Comment (trueadm) [inline packages/react-dom/src/client/ReactDOMComponent.js]: Typo here, you meant to use the function rather than `enableFlareAPI`

**Comment by gaearon** [comment]
Thank you for the effort, but let's not do this.

We generally avoid purely stylistic refactors. They often introduce subtle bugs when some branch was a bit different but no one noticed.

The repetition here makes the code flexible. It's easy to move things around, add or remove or negate some conditions. As soon as they get abstracted into helpers, the code becomes harder to change. You may start having some conditions that never fire, but you might not realize that because they're behind an indirection.

This also starts to mess with dead code elimination. It's harder to predict which code gets included and which doesn't.

**Comment by jamesgeorge007** [comment]
@trueadm @gaearon could you point me an appropriate one to work upon?

**Pull request closed without merging**